src/openforms/api/fields.py

```diff
@@ -1,7 +1,20 @@
-from rest_framework import relations, serializers
+from rest_framework import fields, relations
 
+from openforms.formio.validators import variable_key_validator
 
-class PrimaryKeyRelatedAsChoicesField(serializers.PrimaryKeyRelatedField):
+
+class FormioVariableKeyField(fields.CharField):
```

Comment:
IMO this belongs in either `openforms.formio.api.fields` ~~or `openforms.variables.api.fields`~~, not in the `openforms.api` "framework" code.

Considering the validator is implemented in the `formio` package, the field belongs there too. It's a code smell that you have to import something from a specific module into a general purpose module :nose:

Reply:
Misread https://github.com/open-formulieren/open-forms/pull/3891#discussion_r1491337635